Backend/src/main/java/lt/vu/ads/controllers/OrderController.java

```diff
@@ -21,12 +20,27 @@ public List<Order> getAllOrders(){
 
     @GetMapping("/order/{id}")
     public ResponseEntity < Order > getOrderById(@PathVariable(value = "id") Long orderId)
-            throws ResourceNotFoundException {
+            throws Exception {
         Order order = orderRepository.findById(orderId)
-                .orElseThrow(() -> new ResourceNotFoundException("Order not found for this id :: " + orderId));
+                .orElseThrow(() -> new Exception("Order not found for this id :: " + orderId));
         return ResponseEntity.ok().body(order);
     }
 
+    @PutMapping("/order/{id}")
+    public ResponseEntity < Order > updateOrder(@PathVariable(value = "id") Long orderId,
+                                                       @RequestBody Order orderDetails) throws Exception {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new Exception("Order not found for this id :: " + orderId));
+
+        if(order.getConvenientArrivalTimeTo().before(order.getConvenientArrivalTimeFrom())) {
```

Comment:
I think if one of these values is null exception will be thrown. So i think you should check if values are not null at first